### ISSUE TITLE
Images larger than 3 MB can be uploaded as profile photos

### ISSUE THREAD

**Issue by ioanarusiczki** [mozilla/addons-server#7731]
STR:
1. Load AMO dev and go to Edit profile
2. Upload for the Profile photo an image larger than 3 MB (I tried with 3.78MB, 3.91MB)
3. Click update account
4. Upload again an image larger than 4 MB
5. Click update account and observe the screen

Expected result:
After step 3 Please use images smaller than 3MB and Errors found message are displayed.

Actual results:
Only after a file larger than 4 MB is uploaded the Please use images smaller than 3MB is displayed.

Notes:
If images up to 4 MB are uploaded then the actual message is not correct concerning the size accepted for upload.
This issue is reproducible on AMO dev and AMO stage with FF58 (Win10). 

![3mbissue](https://user-images.githubusercontent.com/33448286/37037012-0ac86062-215a-11e8-8e5c-da90861f6fc7.gif)

**Comment by diox** [comment]
The correct size is definitely 4, not 3. For some reason the code does:
```python
         if photo.size > settings.MAX_PHOTO_UPLOAD_SIZE:
             msg = ugettext('Please use images smaller than %dMB.')
             size_in_mb = settings.MAX_PHOTO_UPLOAD_SIZE / 1024 / 1024 - 1
             raise forms.ValidationError(msg % size_in_mb)
```

Which is causing the value to be rounded down to 3...

**Comment by rafspiny** [comment]
@diox beside the trivial fix, should a test be created for this one? and if so, do you think of a way to test it without using a fixtures with a 4.xxMB png? Mocking is not really usable unless the piece of code you highlighted is refactored I suppose.

**Comment by diox** [comment]
Mocking the photo object should work ?

**Comment by eviljeff** [comment]
Note this is the legacy frontend and right now the replacement edit account profile page is being worked on for the new frontend.  So unless it's an easy fix I wouldn't bother because the code will be deprecated within weeks and deleted within months.

**Comment by deepanshu-jain1999** [comment]
I want to contribute on this issue but I don't know the procedure . I am new in open source.
 Please help

**Comment by eviljeff** [comment]
@deepanshu-jain1999 you should choose another issue - @rafspiny was already looking into it.

**Comment by rafspiny** [comment]
yep @eviljeff 
I finally had some free time to look into Mock and I came up with a test case.
It was a good exercise and I hope it is ok.
Pull request at: https://github.com/mozilla/addons-server/pull/7836

**Comment by ioanarusiczki** [comment]
This issue is verified fixed on AMO dev with FF59 (Win 10)

![4 mb](https://user-images.githubusercontent.com/33448286/37605893-de4ddcd6-2b9c-11e8-8dd0-179e26949d09.png)

**Comment by pstephenwille** [comment]
lame-o!
